Software/lib/GNSE_HAL/GNSE_hal.c

```diff
@@ -0,0 +1,155 @@
+/** Copyright © 2021 The Things Industries B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file GNSE_hal.c
+ *
+ * @copyright Copyright (c) 2021 The Things Industries B.V.
+ *
+ */
+
+#include "GNSE_hal.h"
+#include "GNSE_acc.h"
+#include "GNSE_bm.h"
+#include "SHTC3.h"
+#include "mxic_hc.h"
+#include "BUZZER.h"
+
+/**
+  * @brief Initialises the internal sensors for GNSE, the accelerometer and humidity/temperature sensor
+  * @note I2C interface should be initialised before this function
+  * @param none
+  * @return none
+  */
+void GNSE_Internal_Sensors_Init(void)
+{
+    GNSE_BSP_LS_Init(LOAD_SWITCH_SENSORS);
+    GNSE_BSP_LS_On(LOAD_SWITCH_SENSORS);
+    /* Delay required to let the sensors start up */
+    HAL_Delay(100);
```

Comment:
Please make this a configurable value in GNSE_hal.h, call it `#define GNSE_HAL_INIT_DELAY`

Reply:
and the value 100 is not required, we actually need to reduce it by measuring the settling time of the load switch

Reply:
Shall I create an issue for calculating the settling time?

Reply:
@marnixcro Yeah that would be great, but for this PR 
Just add a random value and link it to the issue, something like 
```c
#define GNSE_HAL_INIT_DELAY 100 // TODO: Decrease delay, see <link to issue>
```